ansible/roles/jenkins-worker/tasks/main.yml

```diff
@@ -84,9 +84,8 @@
   when: "'rhel7-s390x' in inventory_hostname"
 
 - name: Firewall | remove firewalld
-  when: has_firewalld.rc == 0
   raw: yum remove -y firewalld
-  when: "'rhel7-s390x' in inventory_hostname"
+  when: "'rhel7-s390x' in inventory_hostname and has_firewalld.rc == 0"
```

Comment:
What is the behaviour for duplicate keys? If its "last key wins", then this isn't a refactor, its adding a condition. Did you run it  on a host to confirm it works?

Reply:
I don't know what the behaviour is for duplicate keys but the linting is flagging it. I've not run it. Is yum happy to remove something that isn't installed (`has_firewalld.rc` is set in the `Firewall | check for firewalld` check immediately above)?

Reply:
```
% ssh test-ibm-rhel7-s390x-1 sudo yum remove -y firewalld && echo YES
Loaded plugins: product-id, search-disabled-repos, subscription-manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
No Match for argument: firewalld
No Packages marked for removal
YES
```

Reply:
So do we need the `check for firewalld` at all then?

Reply:
It looks suspiciously unnecessary to me. If its last-key wins, the stat resiult was probably always being ignored. Hard to know without trying, though.

Reply:
Not 100% sure but it seems that last key _does_ win https://github.com/ansible/ansible/pull/56933/files#diff-c495d684aecaea8f523caa00ab4ed915R74-R75